Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java

```diff
@@ -237,7 +238,12 @@ private String generateLongUrlWithParams(String baseUrl) {
         if (params != null && params.length() > 0) {
             byte[] data = params.getBytes();
             String base64Data = Base64.encodeToString(data, android.util.Base64.NO_WRAP);
-            longUrl = longUrl + "source=android&data=" + base64Data;
+            try {
+                String urlEncodedBase64Data = URLEncoder.encode(base64Data,"US-ASCII");
+                longUrl = longUrl + "source=android&data=" + urlEncodedBase64Data;
+            } catch (Exception e) {
+                callback_.onLinkCreate(null, new BranchError("Trouble creating a URL.",BranchError.ERR_BRANCH_INVALID_REQUEST));
```

Comment:
Since ` Exception e` is not used please change it to `Exception ignore` to fix the warnings